server/svix-server/src/v1/endpoints/message.rs

```diff
@@ -245,12 +254,28 @@ async fn create_message(
         cache,
         ..
     }): State<AppState>,
-    ValidatedQuery(CreateMessageQueryParams { with_content }): ValidatedQuery<
-        CreateMessageQueryParams,
-    >,
-    permissions::OrganizationWithApplication { app }: permissions::OrganizationWithApplication,
+    Path(ApplicationPath { app_id }): Path<ApplicationPath>,
+    ValidatedQuery(CreateMessageQueryParams {
+        with_content,
+        ignore_missing_app,
+    }): ValidatedQuery<CreateMessageQueryParams>,
+    permissions::Organization { org_id }: permissions::Organization,
     ValidatedJson(data): ValidatedJson<MessageIn>,
-) -> Result<(StatusCode, Json<MessageOut>)> {
+) -> Result<(StatusCode, axum::response::Response)> {
```

Comment:
I think it would be cleaner to have it as an enum rather than either? (assuming enum works with axum)
So an enum for Ok and Accepted and then strict typing inside?

Reply:
Enum is an option, axum doesn't really care either way, as long as it's serializable JSON it's all the same. It does not make a difference to aide either, it can't infer that one status code is one variant of the enum and the other another, it has to be manually told with the operation transformation (the other thing you commented on).

Using `Response` + `into_response()` is their recommendation: https://docs.rs/axum/latest/axum/response/index.html#returning-different-response-types

Do you think the small added extra type safety is worth it for the enum?

Reply:
I'd think so, yeah. It's not a lot of work, and we want to have type checking where we can.